### ISSUE TITLE
Temtem tracker only work once time

### ISSUE THREAD

**Issue by vxtrlx** [mculig/TemtemTracker#14]
i open the temtem tracker, and only work once time when i opened it. i need restart the tracker every time when i go into a battle. I have checked different graphic options and nothing. ty for help

**Comment by vxtrlx** [comment]
The first encounter works for me, and it counts for the tracker, but if I keep looking for fights, the tracker stops working and only the first fight I do is registered once the tracker is opened

**Comment by mculig** [comment]
This happens if the minimap is obstructed OR the aspect ratio of your game client is unsupported or wrongly configured (i.e. playing at a 16:10 aspect ratio on a 16:9 monitor which yes, happens to people for some reason). 

I've also seen this or outright non-detection happen in game clients where the colors of UI elements were changed due to things like running ReShade or other external changes to game graphics.

Finally, I have seen sufficient people running Temtem on Linux and reporting issues despite the game not being supported on the platform to also want to mention that TemtemTracker is untested on Linux and I have no intention of offering support for Linux.

Basically, this is an issue that has something to do with how things are running on your end and not with the code of the tracker itself.

**Comment by vxtrlx** [comment]
and can it be fixed? I would like to have the tracker functional

**Comment by mculig** [comment]
> and can it be fixed? I would like to have the tracker functional

It can. Don't cover the minimap, check your settings. Make sure you're running at an aspect ratio and resolution your monitor actually supports and that that aspect ratio is actually supported by the tracker.

This is on your end, meaning the problem is with something you're doing and not something I can fix.

**Comment by vxtrlx** [comment]
![image](https://user-images.githubusercontent.com/87710592/126347502-4bda91a8-1dc1-4865-89f9-444f54265ed7.png)
i cant fix it, the monitor configuration its ok, in game settings are ok, and i dont know whats happening

**Comment by vxtrlx** [comment]
![image](https://user-images.githubusercontent.com/87710592/126347924-5b0d8e71-a004-437e-8af5-a9744a90646d.png)
i have the temtracker where it dos not cover nothing

**Comment by vxtrlx** [comment]
may it be a nvidia control panel problem?

**Comment by mculig** [comment]
The screenshot you sent is not 1920x1080 resolution. I am getting 1911x1075. If you did not crop any of it away then something is wrong with the resolution you got set up.

The radar UI may also be interfering with minimap detection, but I have not had a reported case of this yet and with your screenshot not being 1920x1080 I can't compare to my detection spots to be sure.

**Comment by vxtrlx** [comment]
![image](https://user-images.githubusercontent.com/87710592/126350548-69f58f0c-0b20-4a83-a66b-30169a858bdf.png)
this is a monitor screen whit the keyboard. I thought I had the right resolution. And you may be right about the radar, I'm going to finish it, and see if it works for me later. Thank you very much for the tip

**Comment by mculig** [comment]
What might help right now is if you open the TemtemTracker folder, then open config, and open the config.json file. Scroll down until you find the "aspectRatios" part. There, find a block starting with "aspectRatio":"16:9" and go down inside that block until you find spot8HeightPercentage. Replace the value there (should be 0.2083) with 0.2037. You can go as low as 0.2 and should be fine. That'll move spot 8 a bit higher and may help avoid the radar UI if it's being clipped in your case.

![image](https://user-images.githubusercontent.com/27305153/126351370-d31b85cb-dc6f-4a99-b4a2-ab459b327d1d.png)

**Comment by mculig** [comment]
Also radars do not work in Saipark.

**Comment by vxtrlx** [comment]
Thank you so much, the problem is the aspect ratio. ty for all the help man!!!! it works very nice now

**Comment by mculig** [comment]
> Thank you so much, the problem is the aspect ratio. ty for all the help man!!!! it works very nice now

The aspect ratio in your game settings, or changing the config?

**Comment by vxtrlx** [comment]
changing the config in config.json

**Comment by vxtrlx** [comment]
it works very well now, ty so much bro

**Comment by mculig** [comment]
Okay. Gonna reopen this as a bug then seeing as it's the radar causing it. Gonna have to move spot 8 away by a bit to give it space on all aspect ratios :(

**Comment by kev494** [comment]
I have a similar issue. It tracks exactly after opening the tracker.
But I have a 21:9 resolution (3440x1440) and I saw in the supported resolution that it is not separately noted only for as format note for others. Does that mean it is not really supported? 
The game itself doesn't seem to use the full 21:9 though, there are some black lines left and right. 
![20230301194103_1](https://user-images.githubusercontent.com/25420055/222234909-9ed041d7-3416-427d-a094-ad94a9bc9200.jpg)

Thanks for your feedback, would love to use this amazing project.

**Comment by mathiaspiehl** [comment]
still got that bug. It works once and then never again, until you restart it.... I also changed resolution, aspect ration and other things, but it doesnt work on a 21:9 Monitor with 3440x1440 at all it seems.... unfortunately this issue exists for 4 years now.

**Comment by mculig** [comment]
> still got that bug. It works once and then never again, until you restart it.... I also changed resolution, aspect ration and other things, but it doesnt work on a 21:9 Monitor with 3440x1440 at all it seems.... unfortunately this issue exists for 4 years now.

Hello. 

While I am sympathetic to the frustrations of finding something you wish to use only for it not to work, as well as the reason being a long-term unaddressed issue, I feel the need to point out a few things.

Firstly, if you take the time to look at the Readme, you will find a Farewell, committed roughly 3 years ago, stating the fact that this repository will no longer be maintained. 

Second, the use of phrases like "unfortunately this issue exists for 4 years now" is, in my opinion, poor manners when interacting with the open source community. While I understand your frustration, I would like to point out that you are commenting on an open source project, provided as-is, free of charge, with no promises or guarantees. Your phrasing may simply be unfortunate, or a manifestation of frustration, but in the future please consider the fact that such phrasing may come across as insulting, demanding or entitled. In the future, please consider the fact that long-standing issues do not exist in a vaccum, but rather as a result of a combination of various factors, such as the scale of the issue, a lack of resources to address it, a lack of demand, or because a repository is no longer being maintained.

Fortunately for you, this is an open source project, so you are free to fork it and fix the issue in your fork, download the source and simply fix it for yourself locally, or take a look at the source and attempt to figure out the root of the problem.

I'll give you some tips to get you started:

In general, the detection loop failing like this means that the end of an encounter has not been correctly identified, thus the next encounter is not triggering the detection loop.

Temtemtracker uses points in the game window and their colors to detect the presence or absence of certain GUI elements to indicate combat/out-of-combat situations. The RGB color values of these pixels as well as their locations as a percentage of screen width or height are located in the config.json config file:

https://github.com/mculig/TemtemTracker/blob/master/TemtemTracker/config/config.json

You may have some luck tweaking these values to adjust for changes in how the game displays colors or changes in the shape and location of graphical assets in the game. 

Best of luck, and please be kinder when interacting with open source projects in the future. Many people do this stuff as volunteers, and deserve only kindness and gratitude.
